pyanaconda/ui/tui/spokes/user.py

```diff
@@ -130,11 +126,31 @@ def status(self):
         else:
             return _("User %s will be created") % self.data.user.userList[0].name
 
+    def input(self, args, key):
+        self.dialog.wrong_input_message = None
+        try:
+            field = self.visible_fields[int(key)-1]
+        except (ValueError, IndexError):
+            pass
+        else:
+            if field.attribute == "gecos":
+                self.dialog.wrong_input_message = _("Full name can't contain ':' character")
+            elif field.attribute == "name":
+                # more granular message is returned by check_name
+                pass
+            elif field.attribute == "_groups":
+                self.dialog.wrong_input_message = _("Either some group name is invalid or groups are not separated by comma")
+
+
+        return EditTUISpoke.input(self, args, key)
+
     def apply(self):
         if self.args.gecos and not self.args.name:
             username = guess_username(self.args.gecos)
-            if not USERNAME_VALID.match(username):
-                self.errors.append(_("Invalid user name: %s.\n") % username)
+            valid, msg = check_name(username)
+            if not valid:
+                self.errors.append(_("Invalid user name: %(username)s.\n%(error_message)s")
+                        % {"username": username, "error_message": msg})
```

Comment:
This is setting of the style_guide check, but it is valid. Either %(username) can be changed to something else, of you can add ```'pyanaconda/ui/tui/spokes/user.py': {'username': 1}``` to expected_badness in tests/gettext/style_guide.py